src/main/java/seedu/edrecord/model/module/ModuleSet.java

```diff
@@ -0,0 +1,150 @@
+
+    /**
+     * Creates a new ModuleSet, which holds a student's modules and classes.
+     */
+    public ModuleSet() {}
+
+    /**
+     * Adds a module to the person's module set.
+     * The module should not have any classes
+     */
+    public void add(Module mod, Group group) {
+        if (!containsModule(mod)) {
+            modules.add(mod);
+        }
+
+        if (!containsGroupInModule(mod, group)) {
+            for (Module module : modules) {
+                if (module.equals(mod)) {
+                    module.addGroup(group);
+                }
+            }
+        }
+    }
+    /**
+     * Adds a module without adding to the module's group system.
+     * Used for testing purposes only.
+     */
+    public void addToSet(Module mod) {
+        modules.add(mod);
+    }
+
+    public void addAll(ModuleSet mods) {
+        modules.addAll(mods.getModules());
+    }
+
+    /**
+     * Removes the given mod based on correct code. The module does not need to have the same classes.
+     * containsModule should be called together with this method.
+     */
+    public void removeMod(Module mod) {
+        for (Module module : this.getModules()) {
+            if (module.getCode().equals(mod.getCode())) {
+                modules.remove(module);
+                return;
+            }
+        }
+    }
+
+    /**
+     * Removes if the module and group is within this module set.
+     * containsModule and containsGroup should be called together with this method.
+     */
+    public void removeGroup(Module mod, Group grp) {
+        for (Module module : this.getModules()) {
+            if (module.getCode().equals(mod.getCode())) {
+                if (module.hasGroup(grp)) {
+                    module.deleteGroup(grp);
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * @return A Set of Modules.
+     */
+    public Set<Module> getModules() {
+        return modules;
+    }
+
+    /**
+     * Checks if the given module is in the set based on correct code.
+     * The module does not need to have the same classes.
+     * @return boolean to show if the set contains the given module.
+     */
+    public boolean containsModule(Module mod) {
+        for (Module module : this.getModules()) {
+            if (module.getCode().equals(mod.getCode())) {
+                return true;
+            }
+        }
+        return false;
+    }
+    /**
+     * Checks if the module and group is within this module set.
+     * containsModule should be called together with this method.
+     * @return boolean to show if the set contains the given group in given module.
+     */
+    public boolean containsGroupInModule(Module mod, Group grp) {
+        for (Module module : this.getModules()) {
+            if (module.getCode().equals(mod.getCode())) {
+                if (module.hasGroup(grp)) {
+                    return true;
+                }
+            }
```

Comment:
```suggestion
            if (module.getCode().equals(mod.getCode()) && module.hasGroup(grp)) {
                return true;
            }
```
Less arrow-head style code